spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/ssh/PropertyBasedSshSessionFactory.java

```diff
@@ -193,6 +179,19 @@ protected Iterable<KeyPair> getDefaultKeys(File dir) {
 		return new SingleKeyIdentityProvider(sshKeysByHostname);
 	}
 
+	private static JGitEnvironmentProperties findEnvironmentProperties(
+			Map<String, JGitEnvironmentProperties> sshKeysByHostname, InetSocketAddress socketAddress) {
+		if (sshKeysByHostname.containsKey(socketAddress.getHostString())) {
+			return sshKeysByHostname.get(socketAddress.getHostString());
+		}
+
+		if (socketAddress.getAddress() != null) {
+			return sshKeysByHostname.get(socketAddress.getAddress().getHostAddress());
+		}
+
+		return null;
```

Comment:
Should we return an empty map to avoid future NPEs?

Reply:
the return value is `JGitEnvironmentProperties`, not a map.  If we return a default instance of `JGitEnvironmentProperties` would cause other issues. the `null`  return value implies that there has been no configuration at all for this host/ip but a default `JGitEnvironmentProperties` does not convey the same message. 

I could use Optional here but since the method is private, I thought the use of Optional is overkill.